HTTP. Работа с сервером

Nitraatss opened the pull request:
---
:mortar_board: [HTTP. Работа с сервером](https://up.htmlacademy.ru/vue/1/user/334301/tasks/6)

Nitraatss committed:
Fixes from Slack

Nitraatss committed:
Update Contributing.md

Nitraatss committed:
5.7. HTTP. Работа с сервером

Djaler commented:
Не отображаются имена ингредиентов в билдере

Djaler commented on src/frontend/src/modules/builder/components/BuilderIngredientsSelector.vue:
```diff
@@ -31,7 +31,10 @@
+                .slice(0, -4)}`"
               :draggable="ingredient.value < 3"
               @dragstart="onDragStart($event, ingredientIndex)"
             >
```
потому что тут в тексте дергается text, а на название лежит в name

Djaler commented:
Немного поехала верстка
![image](https://user-images.githubusercontent.com/7964583/139118571-26470319-53c4-4d02-abf9-1191209e4faf.png)

Djaler commented:
Данные для билдера запрашиваются 2 раза, если перезагрузить страницу истории заказов:
Дергается Buildr/fetchBuilder из init в store, и ещё дёргается Orders/getOrdersList

Djaler commented:
Аналогично, Cart/fetchMisc дергается в init, и ещё раз дёргается в Orders/getOrdersList

Djaler commented:
При заказе на существующий адрес должно быть нельзя редактировать подставленные поля

Djaler commented:
При заказе на существующий адрес должен отправляться `address: { id: идентификатор адреса }` вместо полей адреса, потому что этот адрес сохраняется ещё раз

Djaler commented on src/frontend/src/components/RadioButton.vue:
```diff
@@ -1,5 +1,5 @@
 <template>
-  <label :class="radioClass" @click="onRaidoButtonClick">
+  <label :class="className" @click="onRaidoButtonClick">
```
если нужно установить класс на корневой элемент компонента, можно не передавать его через пропсы, а просто установить класс с помощью `class` на компонент и всё

Djaler commented on src/frontend/src/modules/builder/components/BuilderDoughSelector.vue:
```diff
@@ -6,7 +6,9 @@
         v-for="(dough, doughIndex) in preparedDough"
         :key="`dough-${doughIndex}`"
-        :className="`dough__input dough__input--${dough.value}`"
+        :className="`dough__input dough__input--${
```
такую логику лучше перенести в computed/methods

Djaler commented on src/frontend/src/modules/builder/components/BuilderIngredientsSelector.vue:
```diff
@@ -31,7 +31,10 @@
               class="filling"
-              :class="`filling--${ingredient.name}`"
+              :class="`filling--${ingredient.image
+                .split('/')
```
аналогично, в темплейте такое лучше не держать

Djaler commented on src/frontend/src/modules/orders/components/OrdersSheet.vue:
```diff
@@ -0,0 +1,153 @@
+          <button
+            type="button"
+            class="button button--border"
+            @click="onDeleteButtonClick($event, orderInfo.id)"
```
в этих методах event не используется, зачем туда его передавать?

Djaler commented on src/frontend/src/store/index.js:
```diff
@@ -17,7 +18,9 @@ export default new Vuex.Store({
-      dispatch("Cart/fetchMisc");
+      await dispatch("Auth/getMe");
+      await dispatch("Builder/fetchBuilder");
+      await dispatch("Cart/fetchMisc");
```
хорошо бы эти запросы делать параллельно, чтобы не ждать лишнее время

Djaler commented on src/frontend/src/store/modules/cart.store.js:
```diff
@@ -27,14 +27,29 @@ export default {
+      const pizzaSchema = rootState.Builder.pizzaSchema;
+
+      if (rootState.Auth.isAuthenticated === true) {
+        await dispatch("Address/getAddressList", null, { root: true });
```
не совсем понятно, почему это находится в fetchMisc

Djaler commented on src/frontend/src/store/modules/cart.store.js:
```diff
@@ -27,14 +27,29 @@ export default {
+      if (rootState.Auth.isAuthenticated === true) {
+        await dispatch("Address/getAddressList", null, { root: true });
+
+        commit(UPDATE_CART_FORM, {
```
и это

Nitraatss committed:
Исправление стилей

Nitraatss committed:
Правки

Исправлена двойная подгрузка данных
Модификаторы классов перенесены в methods
Исправлена отправка заказа на существующий адрес
Исправлено отображение имен ингредиентов в конструкторе
Добавлен модификатор количества ингредиентов на превью пиццы в конструкторе

Nitraatss committed:
Оптимизация геттеров

Djaler commented:
> Не отображаются имена ингредиентов в билдере

Не исправлено

Djaler commented on src/frontend/src/store/modules/cart.store.js:
```diff
@@ -50,49 +52,86 @@ export default {
-            console.log("Оформлен заказ с самовывозом");
-          }
+          postData.address = null;
+          await this.$api.orders.post(postData);
```
при выбранном самовывозе заказ делается 2 раза

Djaler commented on src/frontend/src/store/modules/builder.store.js:
```diff
@@ -8,111 +6,30 @@ import {
+      const dough = await this.$api.dough.query();
+      const ingredients = await this.$api.ingredients.query();
+      const sauces = await this.$api.sauces.query();
+      const sizes = await this.$api.sizes.query();
```
тут тоже параллельно запросы стоит делать, а дожидаться всех вместе с помощью Promise.all

Nitraatss committed:
Правки

Исправлено отображение ингредиентов
Исправлено получение информации об ингредиентах
Исправлена двойная отправка заказа при самовывозе
